### ISSUE TITLE
[Bug]: Load segment with DiskANN index may panic

### ISSUE THREAD

**Issue by yah01** [milvus-io/milvus#21787]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

2023-01-18T10:52:55.052309599+08:00 stderr F fatal error: unexpected signal during runtime execution
2023-01-18T10:52:55.054548826+08:00 stderr F [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x7f468f9a0f89]
2023-01-18T10:52:55.054566898+08:00 stderr F 
2023-01-18T10:52:55.054572245+08:00 stderr F runtime stack:
2023-01-18T10:52:55.054576594+08:00 stderr F runtime.throw({0x375004f?, 0x7f468faf1be0?})
2023-01-18T10:52:55.054580382+08:00 stderr F 	/usr/local/go/src/runtime/panic.go:992 +0x71 fp=0x7f442c766b60 sp=0x7f442c766b30 pc=0x11f8151
2023-01-18T10:52:55.054587433+08:00 stderr F runtime.sigpanic()
2023-01-18T10:52:55.054599277+08:00 stderr F 	/usr/local/go/src/runtime/signal_unix.go:802 +0x3a9 fp=0x7f442c766bb0 sp=0x7f442c766b60 pc=0x120ed89
2023-01-18T10:52:55.05460527+08:00 stderr F 
2023-01-18T10:52:55.054609046+08:00 stderr F goroutine 54069 [syscall]:
2023-01-18T10:53:01.256273864+08:00 stderr F non-Go function
2023-01-18T10:53:01.256314921+08:00 stderr F 	pc=0x7f468f9a0f89
2023-01-18T10:53:01.256332101+08:00 stderr F non-Go function
2023-01-18T10:53:01.256335732+08:00 stderr F 	pc=0x7f468f5d61e4
2023-01-18T10:53:01.256350634+08:00 stderr F non-Go function
2023-01-18T10:53:01.256355544+08:00 stderr F 	pc=0x7f468f5dd81b
2023-01-18T10:53:01.256360691+08:00 stderr F runtime.cgocall(0x2e4a000, 0xc001492bf8)
2023-01-18T10:53:01.256364727+08:00 stderr F 	/usr/local/go/src/runtime/cgocall.go:157 +0x5c fp=0xc001492bd0 sp=0xc001492b98 pc=0x11c3b7c
2023-01-18T10:53:01.25636988+08:00 stderr F github.com/milvus-io/milvus/internal/querynodev2/segments._Cfunc_AppendIndex(0x7f448d174180, 0x7f40741a6e00)
2023-01-18T10:53:01.25638477+08:00 stderr F 	_cgo_gotypes.go:172 +0x55 fp=0xc001492bf8 sp=0xc001492bd0 pc=0x2ba2455
2023-01-18T10:53:01.256390068+08:00 stderr F github.com/milvus-io/milvus/internal/querynodev2/segments.(*LoadIndexInfo).appendIndexData.func5(0x7f448d079360?, 0x36fd65e?)
2023-01-18T10:53:01.256395582+08:00 stderr F 	/go/src/github.com/milvus-io/milvus/internal/querynodev2/segments/load_index_info.go:200 +0x65 fp=0xc001492c38 sp=0xc001492bf8 pc=0x2ba7785
2023-01-18T10:53:01.256401409+08:00 stderr F github.com/milvus-io/milvus/internal/querynodev2/segments.(*LoadIndexInfo).appendIndexData(0xc001492dc0?, {0x0, 0x0, 0xc0089a8b88?}, {0xc004840d00, 0x7, 0x7f448d055000?})
2023-01-18T10:53:01.256415277+08:00 stderr F 	/go/src/github.com/milvus-io/milvus/internal/querynodev2/segments/load_index_info.go:200 +0x145 fp=0xc001492ce0 sp=0xc001492c38 pc=0x2ba74a5
2023-01-18T10:53:01.256421841+08:00 stderr F github.com/milvus-io/milvus/internal/querynodev2/segments.(*LoadIndexInfo).appendLoadIndexInfo(0xc003734480?, {0x0, 0x0, 0x0}, 0xc000cc0320, 0x28?, 0x22?, 0x0?, 0xa?)
2023-01-18T10:53:01.256428889+08:00 stderr F 	/go/src/github.com/milvus-io/milvus/internal/querynodev2/segments/load_index_info.go:124 +0x3b5 fp=0xc001492e30 sp=0xc001492ce0 pc=0x2ba6895
2023-01-18T10:53:01.256433374+08:00 stderr F github.com/milvus-io/milvus/internal/querynodev2/segments.(*LocalSegment).LoadIndex(0xc0023ba640, {0x0, 0x0, 0x0}, 0xc000cc0320, 0x0?)
2023-01-18T10:53:01.256437168+08:00 stderr F 	/go/src/github.com/milvus-io/milvus/internal/querynodev2/segments/segment.go:722 +0xf2 fp=0xc001493080 sp=0xc001492e30 pc=0x2bb0c32
2023-01-18T10:53:01.256443291+08:00 stderr F github.com/milvus-io/milvus/internal/querynodev2/segments.(*segmentLoader).loadFieldIndex(0xc000cc2450, {0x3b17eb8?, 0xc0015c6570}, 0xc004f08580?, 0xc000cc0320)
2023-01-18T10:53:01.256448724+08:00 stderr F 	/go/src/github.com/milvus-io/milvus/internal/querynodev2/segments/segment_loader.go:567 +0x525 fp=0xc0014935b0 sp=0xc001493080 pc=0x2b9ac65
2023-01-18T10:53:01.256452369+08:00 stderr F github.com/milvus-io/milvus/internal/querynodev2/segments.(*segmentLoader).loadFieldsIndex(0x335d240?, {0x3b17eb8, 0xc0015c6570}, 0xc0023ba640, 0x1?)
2023-01-18T10:53:01.256456439+08:00 stderr F 	/go/src/github.com/milvus-io/milvus/internal/querynodev2/segments/segment_loader.go:508 +0x11b fp=0xc001493728 sp=0xc0014935b0 pc=0x2b9a4bb
2023-01-18T10:53:01.256465218+08:00 stderr F github.com/milvus-io/milvus/internal/querynodev2/segments.(*segmentLoader).loadSegment(0xc000cc2450, {0x3b17eb8, 0xc0015c6570}, 0xc0023ba640, 0xc0014c43c0)
2023-01-18T10:53:01.256502485+08:00 stderr F 	/go/src/github.com/milvus-io/milvus/internal/querynodev2/segments/segment_loader.go:329 +0xe9e fp=0xc001493bf0 sp=0xc001493728 pc=0x2b9851e
2023-01-18T10:53:01.256512751+08:00 stderr F github.com/milvus-io/milvus/internal/querynodev2/segments.(*segmentLoader).Load.func2(0x101010040?)
2023-01-18T10:53:01.256517742+08:00 stderr F 	/go/src/github.com/milvus-io/milvus/internal/querynodev2/segments/segment_loader.go:175 +0xdc fp=0xc001493dc0 sp=0xc001493bf0 pc=0x2b9621c
2023-01-18T10:53:01.256522989+08:00 stderr F github.com/milvus-io/milvus/internal/util/funcutil.ProcessFuncParallel.func3(0x0)
2023-01-18T10:53:01.256536838+08:00 stderr F 	/go/src/github.com/milvus-io/milvus/internal/util/funcutil/parallel.go:83 +0x16d fp=0xc001493fc8 sp=0xc001493dc0 pc=0x1dd9bad
2023-01-18T10:53:01.256547138+08:00 stderr F github.com/milvus-io/milvus/internal/util/funcutil.ProcessFuncParallel.func4()
2023-01-18T10:53:01.256551498+08:00 stderr F 	/go/src/github.com/milvus-io/milvus/internal/util/funcutil/parallel.go:101 +0x2a fp=0xc001493fe0 sp=0xc001493fc8 pc=0x1dd9a0a
2023-01-18T10:53:01.256555782+08:00 stderr F runtime.goexit()
2023-01-18T10:53:01.256566287+08:00 stderr F 	/usr/local/go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc001493fe8 sp=0xc001493fe0 pc=0x122d0a1
2023-01-18T10:53:01.25657189+08:00 stderr F created by github.com/milvus-io/milvus/internal/util/funcutil.ProcessFuncParallel
2023-01-18T10:53:01.256576367+08:00 stderr F 	/go/src/github.com/milvus-io/milvus/internal/util/funcutil/parallel.go:70 +0x1aa

**Comment by yah01** [comment]
https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20HA%20CI/detail/PR-21625/88/pipeline

**Comment by yah01** [comment]
This may be caused by concurrently loading segments with DiskANN index, QueryNodeV2 loads segments with higher concurrent level, and this problem occurs more often @xige-16

**Comment by yah01** [comment]
After adding more logs, I found that this panic occurs in the `index_->Prepare()`, please help @liliu-z

**Comment by liliu-z** [comment]
talked offline, this is on dev branch. @xige-16 This looks very similar to our known issue(not 100% sure), can you help on pre-screening. Thanks!

**Comment by liliu-z** [comment]
Caused by jemalloc. Jemalloc didn't implement `malloc_trim`, which get called in DiskANN load function. So mixed using of jemalloc's and malloc leads to unpredictable error. This get fixed in Knowhere 1.3.8
